### PR TITLE
IOS-2091 First iteration of new type picker module

### PR DESCRIPTION
**Pull request by ignatovv**
Not in final design, not yet support all the functions but already working.

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private func typesView(types: [ObjectType]) -> some View {
+        ForEach(types) { type in
+            Button {
+                viewModel.didSelectType(type)
+            } label: {
+                HStack {
+                    AnytypeText("\(type.iconEmoji.value) \(type.name)", style: .uxTitle2Medium, color: .Text.primary)
+                }
+            }
+            .padding(.vertical, 15)
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchView.swift:53]: check tap area near border

File: Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                text: text,
+                filteringTypeId: nil, //excludedObjectTypeId,
+                shouldIncludeSets: true, //showSetAndCollection,
+                shouldIncludeCollections: true,// showSetAndCollection,
+                shouldIncludeBookmark: true, //showBookmark,
+                spaceId: spaceId
+            )
+            .map { ObjectType(details: $0) }
+        } catch let error {
+            anytypeAssertionFailure(
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift:26]: not needed. all errors in commands automatically send to sentry

File: Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import Services
+import AnytypeCore
+
+
+final class ObjectTypeSearchInteractor {
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift:5]: Really needed interactor?

File: Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                shouldIncludeBookmark: true, //showBookmark,
+                spaceId: spaceId
+            )
+            .map { ObjectType(details: $0) }
+        } catch let error {
+            anytypeAssertionFailure(
+                "Error in searchObjectTypes",
+                info: ["error": error.localizedDescription]
+            )
+            return []
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift:30]: I think, showing an error as an empty state is the view responsibility

File: Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift (hunk 3 above)
Comment (ignatovv) [reply Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchInteractor.swift:5]: Yep, it will be more robust in the near future

**Review by mgolovko — APPROVED** [review]

**Pull request merged**